### PR TITLE
Add getWorkerTaskReachability API

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Add getWorkerTaskReachability API

closes  https://github.com/temporalio/sdk-java/issues/1862

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    this.taskQueueReachability = taskQueueReachability;
+    this.unretrievedTaskQueues = unretrievedTaskQueues;
+  }
+
+  /**
+   * @return A map of Task Queue names to the reachability status of the Build ID on that queue. If
+   *     the value is an empty list, the Build ID is not reachable on that queue.
+   */
+  public Map<String, List<TaskReachability>> getTaskQueueReachability() {
+    return Collections.unmodifiableMap(taskQueueReachability);
```
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java:45]: I think this should be done in the constructor or by the constructor's caller (same with unmodifiable list below)

File: temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  /**
+   * @return a list of any Task Queues that could not be retrieved because the server limits the
+   *     number that can be queried at once.
+   */
+  public List<String> getUnretrievedTaskQueues() {
```
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java:52]:
```suggestion
  public Set<String> getUnretrievedTaskQueues() {
```

Not required as a set, but arguably clearer.

File: temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java:45]: I would prefer it to be here then in the constructor or trust the caller to have used the correct type

File: temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/** Contains information about the reachability of a specific Build ID. */
+public class BuildIdReachability {
+  private final Map<String, List<TaskReachability>> taskQueueReachability;
+  private final List<String> unretrievedTaskQueues;
+
+  public BuildIdReachability(
```
[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java:33]: ah good catch `WorkerTaskReachability` should also be public

**Review by Sushisource — APPROVED** [review]
Thanks for this! I hope this wasn't something I was supposed to go back and add and forgot about :grimacing:

File: temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java (hunk 1 above)
Comment (cretz) [reply temporal-sdk/src/main/java/io/temporal/client/BuildIdReachability.java:46]: I see this is already merged, but curious why not the constructor? Why create a new object every time a getter is called if you don't need to?

**Pull request merged**